tools/ci_build/github/azure-pipelines/templates/c-api-cpu.yml

```diff
@@ -45,7 +45,7 @@ jobs:
     BaseImage: 'centos:7'
     OnnxruntimeArch: 'x64'
     OnnxruntimeCFlags: '-Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -fstack-clash-protection -fcf-protection -O3 -Wl,--strip-all'
-    OnnxruntimeCXXFlags: '-Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -fstack-clash-protection -fcf-protection -O3 -Wl,--strip-all'
+    OnnxruntimeCXXFlags: '-Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -fstack-clash-protection -fcf-protection -O3 -Wno-unused-parameter -Wl,--strip-all'
```

Comment:
why disable this warning?

Reply:
It was to suppress the error of `onnxruntime_src/cmake/external/protobuf/src/google/protobuf/repeated_ptr_field.h:752:66: error: unused parameter ‘prototype’ [-Werror=unused-parameter]`

the source code is
```
  static inline GenericType* NewFromPrototype(const GenericType* prototype,
                                              Arena* arena = nullptr) {
    return New(arena);
  }
```
https://github.com/protocolbuffers/protobuf/blob/a20c65f2cd549445fda907f7b83894c8eb7427d6/src/google/protobuf/repeated_ptr_field.h#L752-L755.

And thank your question. I double checked that it played a duplicated role with https://github.com/microsoft/onnxruntime/pull/14279/commits/a05dc54e4aaf577d3957cf4dd12be305cd509794.
The change could be removed.

Reply:
I think it's better to suppress them in such a way: https://github.com/microsoft/onnxruntime/blob/main/include/onnxruntime/core/common/eigen_common_wrapper.h, so that the impact would be as small as possible.